### PR TITLE
Public static fields should also be final

### PR DESCRIPTION
**Pull request by klaren**
Also removed unnecessary boxing of booleans.

### CODE REVIEW

File: community/command-line/src/main/java/org/neo4j/commandline/admin/AdminTool.java [hunk 1]
```diff
@@ -36,9 +36,9 @@
     public static final int STATUS_SUCCESS = 0;
     public static final int STATUS_ERROR = 1;
     // These fields are mutable for in-process AdminTool testing purposes.
-    public static String NEO4J_HOME = System.getenv().getOrDefault( "NEO4J_HOME", "" );
-    public static String NEO4J_CONF = System.getenv().getOrDefault( "NEO4J_CONF", "" );
-    public static String NEO4J_DEBUG = System.getenv().getOrDefault( "NEO4J_DEBUG", "" );
+    public static final String NEO4J_HOME = System.getenv().getOrDefault( "NEO4J_HOME", "" );
+    public static final String NEO4J_CONF = System.getenv().getOrDefault( "NEO4J_CONF", "" );
+    public static final String NEO4J_DEBUG = System.getenv().getOrDefault( "NEO4J_DEBUG", "" );
```
Comment (chrisvest) [inline community/command-line/src/main/java/org/neo4j/commandline/admin/AdminTool.java:40]: These are mutable for testing, and exploited in the blockdevice repo, I believe.

Comment (chrisvest) [reply community/command-line/src/main/java/org/neo4j/commandline/admin/AdminTool.java:40]: The comment just above also says so.

Comment (klaren) [reply community/command-line/src/main/java/org/neo4j/commandline/admin/AdminTool.java:40]: Well spotted. Although, why can't the affected tests instantiate `AdminTool` directly as all other tests do?

Comment (chrisvest) [reply community/command-line/src/main/java/org/neo4j/commandline/admin/AdminTool.java:40]: Maybe that wasn't possible at that time.

File: enterprise/com/src/main/java/org/neo4j/com/Client.java [hunk 2]
```diff
@@ -208,7 +208,7 @@ public void start()
             protected ChannelContext create()
             {
                 msgLog.info( threadInfo() + "Trying to open a new channel from " + origin + " to " + destination,
-                        true );
+                        Boolean.TRUE );
```
Comment (MishaDemianenko) [inline enterprise/com/src/main/java/org/neo4j/com/Client.java]: we should be able to just remove those true's from log messages in this and other classes that do the same thing

File: community/common/src/main/java/org/neo4j/function/Predicates.java [hunk 3]
```diff
@@ -262,7 +262,7 @@ public static void awaitForever( BooleanSupplier condition, long checkInterval,
         };
     }
 
-    public static IntPredicate ALWAYS_TRUE_INT = v -> true;
+    public static final IntPredicate ALWAYS_TRUE_INT = v -> true;
```
Comment (MishaDemianenko) [inline community/common/src/main/java/org/neo4j/function/Predicates.java]: please move this constant from the middle of the class to the top

**Review by MishaDemianenko — CHANGES_REQUESTED** [review]

**Comment by MishaDemianenko** [comment]
Also, the thing that Chris mentioned if its not a problem anymore comment on top of those fields should be removed

**Comment by chrisvest** [comment]
@MishaDemianenko I agree. Let's remove the comment, make those fields final, and fix up the blockdevice repo.

File: community/command-line/src/main/java/org/neo4j/commandline/admin/AdminTool.java (hunk 1 above)
Comment (klaren) [reply community/command-line/src/main/java/org/neo4j/commandline/admin/AdminTool.java:40]: Should hopefully be possible now, I'll have a look at it.

**Comment by klaren** [comment]
@MishaDemianenko addressed. I'm also looking into blockdevice to fix compilation errors.

**Review by MishaDemianenko — APPROVED** [review]

**Pull request merged**